src/main/java/org/sunbird/searchby/service/SearchByService.java

```diff
@@ -408,4 +412,159 @@ private List<FracCommonInfo> getMasterPositionList(List<String> positionNameList
 		}
 		return positionList;
 	}
+
+	public Collection<CompetencyInfo> listCompetenciesByOrg(String orgId, String userToken) {
+		 //  id's of content (contentSearch for every ids and implement facets)
+		try {
+			List<String> competenciesList = redisCacheMgr.hget(Constants.TOP_COMMENTS_ORG_REDIS_KEY, cbExtServerProperties.getRedisInsightIndex(), orgId);
+			Map<String, CompetencyInfo> competencyMap = new HashMap<>();
+			if (CollectionUtils.isEmpty(competencyMap)) {
+				logger.info("Initializing/Refreshing the Cache Value for Key : " + Constants.COMPETENCY_CACHE_NAME);
+//				competencyMap = listCompetencyDetails(userToken, competenciesList);
+			}
+			return competencyMap.values();
+		}catch (Exception e){
+			logger.error(e);
+		}
+		return null;
+	}
+
+	private Map<String, CompetencyInfo> listCompetencyDetails(String userToken, List<String> identifiers) throws Exception {
+		Map<String, CompetencyInfo> competencyMap;
+		Map<String, List<CompetencyInfo>> comInfoByType = new HashMap<>();
+		Map<String, List<CompetencyInfo>> comInfoByArea = new HashMap<>();
+
+		// Get facets from Composite Search
+		Map<String, String> headers = new HashMap<>();
+		headers.put(Constants.USER_TOKEN, userToken);
+		headers.put(Constants.AUTHORIZATION, cbExtServerProperties.getSbApiKey());
+
+		HashMap<String, Object> reqBody = new HashMap<>();
+		HashMap<String, Object> req = new HashMap<>();
+		req.put(Constants.FACETS, Arrays.asList(Constants.COMPETENCIES_V5+"."+Constants.COMPETENCY_AREA,
+				Constants.COMPETENCIES_V5+"."+Constants.COMPETENCY_THEMES,
+				Constants.COMPETENCIES_V5+"."+Constants.COMPETENCY_SUB_THEMES));
+		Map<String, Object> filters = new HashMap<>();
+		filters.put(Constants.IDENTIFIER, Arrays.asList(filters));
```

Comment:
some confusion ?
this suppose to be identifiers list ?

Reply:
we have planned to move this API in another PR. so removed the code.